test/cpp/end2end/sdk_authz_end2end_test.cc

```diff
@@ -360,6 +397,29 @@ TEST_F(
   EXPECT_TRUE(resp.message().empty());
 }
 
+TEST_F(SdkAuthzEnd2EndTest,
+       StaticInitAllowsRpcRequestWithPrincipalsFieldOnAuthenticatedConnection) {
```

Comment:
We should probably also test that this is *not* allowed on an insecure connection.

Reply:
We have a test "StaticInitDeniesRpcRequestWithPrincipalsFieldOnUnauthenticatedConnection" to check RPC is denied on unauthenticated connection. It is just above this test.